lib/ddr/models/has_struct_metadata.rb

```diff
@@ -11,6 +11,38 @@ module HasStructMetadata
 
         has_attributes :file_group, :file_use, :order,
                        datastream: "structMetadata", multiple: false
+
+        after_create :assign_struct_metadata!
+      end
+
+      def assign_struct_metadata!
```

Comment:
This method is quite complex.  I'd like to explore how we might break it into smaller, easier to comprehend chunks.